pandas/tests/reshape/test_melt.py

```diff
@@ -1014,3 +1014,12 @@ def test_col_substring_of_stubname(self):
         )
         result = pd.wide_to_long(wide_df, stubnames="PA", i=["node_id", "A"], j="time")
         tm.assert_frame_equal(result, expected)
+
+    def test_warn_of_column_name_value(self):
+        # GH34731
+        # raise a warning if the resultant value column name matches
+        # a name in the dataframe already (default name is "value")
+        df = pd.DataFrame({"col": list("ABC"), "value": range(10, 16, 2)})
+
+        with tm.assert_produces_warning(FutureWarning):
+            dfm = df.melt(id_vars="value")  # noqa F841
```

Comment:
can you create an expected value and use assert_frame_equal

also instead of dfm, call this result.